### PR TITLE
[2.5] Use additional CA for hosted providers

### PR DESCRIPTION
**Pull request by thedadams**
Issue:
https://github.com/rancher/rancher/issues/32903

Original PR:
https://github.com/rancher/rancher/pull/32900

### CODE REVIEW

**Review by cmurphy — DISMISSED** [review]

**Review by cmurphy — APPROVED** [review]

**Review by rmweir — DISMISSED** [review]

**Review by mjura — DISMISSED** [review]
LGTM

**Comment by cbron** [comment]
Think this needs updates from other PR, then gtg

File: pkg/controllers/management/gke/gke_cluster_handler.go [hunk 1]
```diff
@@ -417,36 +418,61 @@ func (e *gkeOperatorController) deployGKEOperator() error {
...
-func (e *gkeOperatorController) generateSATokenWithPublicAPI(cluster *mgmtv3.Cluster) (string, bool, error) {
-	var publicAccess bool
-
-	ctx := context.Background()
-	ts, err := controller.GetTokenSource(ctx, e.SecretsCache, cluster.Spec.GKEConfig)
-
-	netDialer := net.Dialer{
+// generateSATokenWithPublicAPI tries to get a service account token from the cluster using the public API endpoint.
+// This function is called if the cluster has only privateEndpoint enabled and is not publicly available.
+// If Rancher is able to communicate with the cluster trough its API endpoint even though it is private, then this function will retrieve
```
Comment (cbron) [inline pkg/controllers/management/gke/gke_cluster_handler.go:423]: nit: 
```suggestion
// If Rancher is able to communicate with the cluster through its API endpoint even though it is private, then this function will retrieve
```

**Review by cbron — APPROVED** [review]
lgtm

File: pkg/controllers/management/gke/gke_cluster_handler.go (hunk 1 above)
Comment (thedadams) [reply pkg/controllers/management/gke/gke_cluster_handler.go]: Fixed.

**Review by cbron — APPROVED** [review]

**Review by cmurphy — APPROVED** [review]

**Pull request merged**